AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/AWSCognitoAuthPlugin+Configure.swift

```diff
@@ -102,6 +102,15 @@ extension AWSCognitoAuthPlugin {
                 configuration.httpClientEngine = .userAgentEngine(for: configuration)
             }
 
+            if let requestTimeout = networkPreferences?.timeoutIntervalForRequest {
+                let requestTimeOutMs = requestTimeout * 1_000
+                configuration.connectTimeoutMs = UInt32(requestTimeOutMs)
+            }
+
+            if let maxRetryUnwrapped = networkPreferences?.maxRetryCount {
+                configuration.retryStrategyOptions = RetryStrategyOptions(maxRetriesBase:  Int(maxRetryUnwrapped))
```

Comment:
Not related to this PR, question for my knowledge, when/where in the plugin validates the 10 limit for retry count?

Reply:
Should _we_ do the validation then?

I guess not because this is network-related, but for example if we know that the service will absolutely fail if we retry more than 10 times, then we might wanna add the validation.

Reply:
Hmmm I kinda disagree with "the service's behaviour may change in the future" statement, on a conceptual basis.

If we know the service fails or does not support a certain thing, I don't think we should still allow that very thing to be attempted. It's like setting ourselves (or Amplify's customers) for failure.

But again, I don't really know the specifics of why this particular use case was implemented in the first place, so I'm fine with the removal of it if you guys think it doesn't apply anymore.